test/models/database/post_repository_test.dart

```diff
@@ -0,0 +1,335 @@
+        throwsA(isA<Exception>()),
+      );
+    });
+  });
+
+  group("Post Firestore testing", () {
+    test("hash overrides correctly", () {
+      const geoPoint = GeoPoint(40, 20);
+      const geoHash = "azdz";
+
+      const location = PostLocationFirestore(
+        geoPoint: geoPoint,
+        geohash: geoHash,
+      );
+
+      final data = PostFirestoreData(
+        ownerId: "owner_id",
+        title: "post_tiltle",
+        description: "description",
+        publicationTime: Timestamp.fromMillisecondsSinceEpoch(4564654),
+        voteScore: 12,
+      );
+
+      final post = PostFirestore(
+        id: "post_id",
+        location: location,
+        data: data,
+      );
+
+      final expectedHash = Object.hash(post.id, post.location, post.data);
+
+      final actualHash = post.hashCode;
+
+      expect(actualHash, expectedHash);
+    });
+  });
+
+  group("Post Repository testing", () {
+    late FakeFirebaseFirestore firestore;
+    late GeoFlutterFire geoFire;
+    late GeoLocationService geoLocationService;
+    late PostRepository postRepository;
+    late MockFirestorePost mockFirestorePost;
+
+    /// Helper function to set a post in the firestore db
+    Future<void> setPostFirestore(PostFirestore post) async {
+      final Map<String, dynamic> locationData = {
+        PostLocationFirestore.geoPointField: post.location.geoPoint,
+        PostLocationFirestore.geohashField: post.location.geohash,
+      };
+
+      await firestore
+          .collection(PostRepository.collectionName)
+          .doc(post.id)
+          .set({
+        PostFirestore.locationField: locationData,
+        ...post.data.toDbData(),
+      });
+    }
+
+    Future<void> setPostsFirestore(List<PostFirestore> posts) async {
+      for (final post in posts) {
+        await setPostFirestore(post);
+      }
+    }
+
+    setUp(() {
+      firestore = FakeFirebaseFirestore();
+      geoFire = GeoFlutterFire();
+      geoLocationService = MockGeoLocationService();
+      postRepository = PostRepository(
+        firestore: firestore,
+        geoFire: geoFire,
+        geoLocationService: geoLocationService,
+      );
+      mockFirestorePost = MockFirestorePost();
+    });
+
+    final post = PostFirestore(
+      id: "post_id",
+      location: const PostLocationFirestore(
+        geoPoint: GeoPoint(40, 20),
+        geohash: "afed",
+      ),
+      data: PostFirestoreData(
+        ownerId: "owner_id",
+        title: "post_tiltle",
+        description: "description",
+        publicationTime: Timestamp.fromMillisecondsSinceEpoch(4564654),
+        voteScore: 12,
+      ),
+    );
+
+    test("delete post correctly", () async {
+      await setPostFirestore(post);
+
+      await postRepository.deletePost(post.id);
+      final actualPost = await firestore
+          .collection(PostRepository.collectionName)
+          .doc(post.id)
+          .get();
+      expect(actualPost.exists, false);
```

Comment:
I would personally find the test more logical if it verifies the post exists before deleting it. Currently, the `setPostFirestore` could not update the database and we would not be able to notice. Adding this verification, we would indeed see that a post which exists, no longer exists after being deleted.

Reply:
Yes, this should make the test more robust, fix in 4a6b6bcb4694b2e67dbebafed8d3199b6c521131.